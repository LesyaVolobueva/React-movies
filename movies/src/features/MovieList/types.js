export const INCLIKE = 'INCLIKE';
export const DECLIKE = 'DECLIKE';
export const OPEN_MOVIE_DETAILS = 'OPEN_MOVIE_DETAILS';
export const RECEIVE_MOVIES = 'RECEIVE_MOVIES';
export const SET_RATING = 'SET_RATING';
export const RECEIVE_ACTORS = 'RECEIVE_ACTORS';
export const DELETE_MOVIE = 'DELETE_MOVIE';
export const SAVE_MOVIE = 'SAVE_MOVIE';
export const SET_LOGGED_IN = 'SET_LOGGED_IN';
