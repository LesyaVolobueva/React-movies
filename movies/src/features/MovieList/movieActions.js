import {
  INCLIKE,
  DECLIKE,
  RECEIVE_MOVIES,
  SET_RATING,
  RECEIVE_ACTORS,
  DELETE_MOVIE,
  SAVE_MOVIE,
  SET_LOGGED_IN,
} from './types';

export const receiveMovies = (movies) => ({
  type: RECEIVE_MOVIES,
  payload: movies,
});

export const fetchMovies = () => {
  return (dispatch, state, api) => {
    return api('movies')
      .then(response => {
        dispatch(receiveMovies(response.data));
      });
  };
};

export const receiveActors = (actors) => ({
  type: RECEIVE_ACTORS,
  payload: actors,
});

export const fetchActors = () => {
  return (dispatch, state, api) => {
    return api('actors')
      .then(response => {
        dispatch(receiveActors(response.data));
      });
  };
};

export const incLike = (id) => ({
  type: INCLIKE,
  payload: id,
});

export const fetchIncLike = (id) => {
  return (dispatch, state, api) => {
    const movie = state().movies.movies.find(movie => movie.id === id);
    return api(`movies/${id}`, 'patch', { ...movie, likes: movie.likes + 1 })
      .then(response => {
        dispatch(incLike(response.data.id));
      });
  };
};

export const decLike = (id) => ({
  type: DECLIKE,
  payload: id,
});

export const fetchDecLike = (id) => {
  return (dispatch, state, api) => {
    const movie = state().movies.movies.find(movie => movie.id === id);
    return api(`movies/${id}`, 'patch', { ...movie, likes: movie.likes - 1 })
      .then(response => {
        dispatch(decLike(response.data.id));
      });
  };
};

export const setRating = (id, stars) => ({
  type: SET_RATING,
  payload: { id, stars },
});

export const fetchSetRating = (id, stars) => {
  return (dispatch, state, api) => {
    const movie = state().movies.movies.find(movie => movie.id === id);
    return api(`movies/${id}`, 'patch', { ...movie, stars })
      .then(() => {
        dispatch(setRating(id, stars));
      });
  };
};

export const deleteMovie = (id) => ({
  type: DELETE_MOVIE,
  payload: id,
});

export const fetchDeleteMovie = (id) => {
  return (dispatch, state, api) => {
    return api(`movies/${id}`, 'delete')
      .then(() => {
        dispatch(deleteMovie(id));
      });
  };
};

export const saveMovie = (movie) => ({
  type: SAVE_MOVIE,
  payload: movie,
});

export const fetchSaveMovie = (movie) => {
  return (dispatch, state, api) => {
    return api(`movies/${movie.id}`, 'patch', { ...movie })
      .then(() => {
        dispatch(saveMovie(movie));
      });
  };
};

