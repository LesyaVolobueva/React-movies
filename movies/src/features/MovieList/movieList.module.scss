.movieList {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 10px 0;
  flex-basis: 100%;
  padding: 10px;
}

.moviesContainer {
  flex-basis: 100%;
}
