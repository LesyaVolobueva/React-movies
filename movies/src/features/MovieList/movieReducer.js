import {
  INCLIKE,
  DECLIKE,
  RECEIVE_MOVIES,
  SET_RATING,
  RECEIVE_ACTORS,
  DELETE_MOVIE,
  SAVE_MOVIE,
  // SET_LOGGED_IN,
} from './types';

export const initialState = {
  movies: [],
  actors: [],
};

export default function movieReducer(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_MOVIES:

      return {
        ...state,
        movies: action.payload,
      };

    case RECEIVE_ACTORS:
      return {
        ...state,
        actors: action.payload,
      };


    case INCLIKE:
      return {
        ...state,
        movies: state.movies.map(movie => movie.id === action.payload
          ? { ...movie, likes: movie.likes + 1 }
          : movie
        ),
      };

    case DECLIKE:
      return {
        ...state,
        movies: state.movies.map(movie => movie.id === action.payload
          ? { ...movie, likes: movie.likes - 1 }
          : movie
        ),
      };

    case SET_RATING:
      return {
        ...state,
        movies: state.movies.map(movie => movie.id === action.payload.id
          ? { ...movie, stars: action.payload.stars }
          : movie
        ),
      };

    case DELETE_MOVIE:
      return {
        ...state,
        movies: [...state.movies].filter(movie => movie.id !== action.payload),
      };

    case SAVE_MOVIE:
      return {
        ...state,
        movies: [...state.movies].map(movie => {
          if (movie.id === action.payload.id) {
            return action.payload;
          }
          return movie;
        }),
      };

    // case SET_LOGGED_IN:
    //   return {
    //     ...state,
    //     isLoggedIn: action.payload,
    //   };


    default:
      return state;
  }
}
