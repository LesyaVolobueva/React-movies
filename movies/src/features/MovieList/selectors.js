export const getFilteredMovies = (state) => {
  const { movies } = state.movies;
  const { isSortedByLikes, isSortedByRating, searchValue } = state.filter;
  let filteredMovies = isSortedByLikes
    ? [...movies].sort((a, b) => b.likes - a.likes)
    : movies;

  filteredMovies = isSortedByRating
    ? [...filteredMovies].sort((a, b) => b.stars - a.stars)
    : filteredMovies;

  filteredMovies = searchValue
    ? filteredMovies.filter(movie => (
      movie.title.toLocaleLowerCase().includes(searchValue)
    ))
    : filteredMovies;

  return filteredMovies;
};

export const findMovie = (state, ownProps) => {
  const movie = Object.assign(
    {},
    state.movies.movies.find(movie => movie.id === Number(ownProps.match.params.id))
  );
  movie.genres = movie.genres.join(', ');
  return movie;
};

export const findActors = (state, ownProps) => {
  const movie = state.movies.movies.find(movie => movie.id === Number(ownProps.match.params.id));
  const actorsIds = movie.actorsIds;
  const actors = [];

  actorsIds.forEach(id =>
    state.movies.actors.forEach(actor => {
      if (actor.id === id) {
        actors.push({ id, name: actor.name });
      }
      return actor;
    })
  );
  return actors;
};

export const getActor = (state, ownProps) => {
  return state.movies.actors.find(actor => actor.id === Number(ownProps.match.params.id));
};

