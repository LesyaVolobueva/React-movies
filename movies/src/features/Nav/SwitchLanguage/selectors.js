import translations from '../../../translation';

export const getLangs = () => {
  const langsList = [];

  for (const lang in translations) {
    langsList.push(lang);
  }
  return langsList;
};
