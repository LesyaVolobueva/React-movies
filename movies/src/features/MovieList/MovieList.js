import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Movie from './Movie/Movie';
import styles from './movieList.module.scss';
import { fetchIncLike, fetchDecLike, setRating } from './movieActions';
import { getFilteredMovies } from './selectors';

const mapStateToProps = (state) => ({
  movies: getFilteredMovies(state),
});

const mapDispatchToProps = (dispatch) => ({
  fetchIncLike: id => dispatch(fetchIncLike(id)),
  fetchDecLike: id => dispatch(fetchDecLike(id)),
  setRating: (id, stars) => dispatch(setRating(id, stars)),
});

const MovieList = (props) => {
  const {
    movies,
    fetchIncLike,
    fetchDecLike,
    setRating,
  } = props;

  return (
    <div className={styles.moviesContainer}>
      <div className={styles.movieList}>
        {movies.map(movie =>
          <Movie
            decLike={fetchDecLike}
            incLike={fetchIncLike}
            key={movie.id}
            movie={movie}
            setRating={setRating}
          />
        )}
      </div>
    </div>
  );
};

MovieList.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      posterUrl: PropTypes.string,
      stars: PropTypes.number,
      likes: PropTypes.number,
      genres: PropTypes.arrayOf(PropTypes.string),
      actors: PropTypes.arrayOf(PropTypes.number),
      director: PropTypes.string,
      description: PropTypes.string,
    })
  ),
  fetchIncLike: PropTypes.func,
  fetchDecLike: PropTypes.func,
  setRating: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MovieList);
