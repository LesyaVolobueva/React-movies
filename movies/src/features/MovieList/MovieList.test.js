import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import MovieList from './MovieList';
import callApi from '../../utils/callApi';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import {
  RECEIVE_ACTORS,
  RECEIVE_MOVIES,
  INCLIKE,
  DECLIKE,
  DELETE_MOVIE,
  SET_RATING,
  SAVE_MOVIE,
} from './types';

import {
  incLike,
  receiveActors,
  receiveMovies,
  decLike,
  deleteMovie,
  saveMovie,
  fetchActors,
  fetchDecLike,
  fetchDeleteMovie,
  fetchIncLike,
  fetchMovies,
  fetchSaveMovie,
  fetchSetRating,
  setRating
} from './movieActions';

import { getFilteredMovies} from './selectors';

import movieReducer, { initialState } from './movieReducer';

const mockStore = configureMockStore([thunk.withExtraArgument(callApi)]);

const lang = {
  edit: 'Edit',
  delete: 'Delete',
  save: 'Save',
  back: 'Go Back',
  signUp: 'Sign Up',
  login: 'Log In',
  movies: 'Movies',
  logOut: 'Log Out',
  imgUrl: 'Img Url',
  title: 'Title',
  director: 'Director',
  genres: 'Genres',
  description: 'Description',
  likes: 'Likes',
  actors: 'Actors',
  searchPlaceholder: 'Enter the name',
  byLikes: 'By likes',
  byRate: 'By rate',
  pleaseLogin: 'Please log in',
  userLogin: 'Your login',
  userPassword: 'Your password',
  passwordIncorrect: 'Password is incorrect!',
  userDoNotExist: 'User doesn\'t exist',
  dontHaveAcc: 'Don\'t have an account?',
  pleaseSignUp: 'Please sign up',
  suchUserExists: 'User with such login exists!',
  alreadyHaveAcc: 'Already have the account?',
};

const actors = [
  {
    "id": 1,
    "name": "Morgan Freeman",
    "imgUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTc0MDMyMzI2OF5BMl5BanBnXkFtZTcwMzM2OTk1MQ@@._V1_UX214_CR0,0,214,317_AL_.jpg",
    "biography": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
  },
  {
    "id": 2,
    "name": "Bob Gunton",
    "imgUrl": "https://i.pinimg.com/736x/9a/69/1e/9a691e7471b7cff69a86262bbcbb2e4b--bob-gunton-epic-film.jpg",
    "biography": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum"
  },
];

const movies = [
  {
    "id": 1,
    "title": "The Shawshank Redemptoin",
    "posterUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SY500_CR0,0,336,500_AL_.jpg",
    "stars": 5,
    "likes": 22,
    "genres": [
      "Crime",
      "Drama"
    ],
    "actorsIds": [
      0,
      1,
      2
    ],
    "director": "Frank Darabont",
    "description": "Chronicles the experiences of a formerly successful banker as a prisoner in the gloomy jailhouse of Shawshank after being found guilty of a crime he did not commit. The film portrays the man's unique way of dealing with his new, torturous life; along the way he befriends a number of fellow prisoners, most notably a wise long-term inmate named Red.",
    "rating": 3
  },
  {
    "id": 4,
    "title": "The Dark Knight",
    "posterUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SY500_CR0,0,337,500_AL_.jpg",
    "stars": 2,
    "likes": 15,
    "genres": [
      "Crime",
      "Drama",
      "Action"
    ],
    "actorsIds": [
      8,
      9,
      10
    ],
    "director": "Nolan",
    "description": "Despite his tarnished reputation after the events of The Dark Knight, in which he took the rap for Dent's crimes, Batman feels compelled to intervene to assist the city and its police force which is struggling to cope with Bane's plans to destroy the city",
    "rating": 5
  },
];

const newMovie = {
  "id": 1,
  "title": "The Shawshank Redemptoin123456",
  "posterUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SY500_CR0,0,336,500_AL_.jpg",
  "stars": 5,
  "likes": 23,
  "genres": [
    "Crime",
    "Drama"
  ],
  "actorsIds": [
    0,
    1,
    2
  ],
  "director": "Frank Darabont",
  "description": "Chronicles the experiences of a formerly successful banker as a prisoner in the gloomy jailhouse of Shawshank after being found guilty of a crime he did not commit. The film portrays the man's unique way of dealing with his new, torturous life; along the way he befriends a number of fellow prisoners, most notably a wise long-term inmate named Red.",
  "rating": 4
};

describe('-- MovieList --', () => {
  describe('VIEW', () => {
    it('renders correctly', () => {
      const store = {
        lang: {
          lang: 'en',
        },
        movies: {
          movies,
        },
        filter: {
          isSortedByLikes: false,
          isSortedByRating: false,
          searchValue: '',
        },
      };

      const props = {
        lang,
        fetchMovies: () => {},
        fetchActors: () => {},
        incLike: () => {},
        decLike: () => {},
        setRating: () => {},
        getFilteredMovies: () => {},
        movies,
        actors,
      };

      const tree = renderer
        .create(
          <MemoryRouter>
            <Provider store={mockStore(store)}>
              <MovieList {...props} />
            </Provider>
          </MemoryRouter>
        );

      expect(tree).toMatchSnapshot();
    });
  });

  describe('ACTIONS', () => {
    let action;

    it('should create receiveActors action', () => {
      action = receiveActors(actors);

      expect(action.type).toBe(RECEIVE_ACTORS);
      expect(action.payload).toBe(actors);
    });

    it('should create receiveMovies action', () => {
      action = receiveMovies(movies);

      expect(action.type).toBe(RECEIVE_MOVIES);
      expect(action.payload).toBe(movies);
    });

    it('should create incLike action', () => {
      action = incLike(movies[0].id);

      expect(action.type).toBe(INCLIKE);
      expect(action.payload).toBe(movies[0].id);
    });

    it('should create decLike action', () => {
      action = decLike(movies[0].id);

      expect(action.type).toBe(DECLIKE);
      expect(action.payload).toBe(movies[0].id);
    });

    it('should create setRating action', () => {
      const stars = 5;
      action = setRating(movies[0].id, stars);

      expect(action.type).toEqual(SET_RATING);
      expect(action.payload).toEqual({
        id: movies[0].id,
        stars,
      });
    });

    it('should create saveMovie action', () => {
      action = saveMovie(newMovie);

      expect(action.type).toEqual(SAVE_MOVIE);
      expect(action.payload).toEqual(newMovie);
    });

    it('should create deleteMovie action', () => {
      action = deleteMovie(movies[0].id);

      expect(action.type).toBe(DELETE_MOVIE);
      expect(action.payload).toBe(movies[0].id);
    });
  });

  describe('ACYNC ACTIONS', () => {
    const mock = new MockAdapter(axios);
    const incLikeMovie = {
      ...movies[0],
      likes: movies[0].likes + 1,
    };
    const decLikeMovie = {
      ...movies[0],
      likes: movies[0].likes - 1,
    };

    mock
      .onGet('http://localhost:3001/movies').reply(200, movies)
      .onGet('http://localhost:3001/actors').reply(200, actors)
      .onDelete('http://localhost:3001/movies/4').reply(200)
      .onPatch('http://localhost:3001/movies/1').reply(200, newMovie)
      .onPatch('http://localhost:3001/movies/1').reply(200, incLikeMovie)
      .onPatch('http://localhost:3001/movies/1').reply(200, decLikeMovie);

    let store;

    beforeEach(() => {
      const data = {
        movies: {
          movies,
          actors,
        },
      };

      store = mockStore(data);
    });

    it('should create fetchActors action', (done) => {
      return store.dispatch(fetchActors())
        .then(() => {
          expect(store.getActions()).toEqual([
            receiveActors(actors)
          ]);
          done();
        });
    });

    it('should create fetchMovies action', (done) => {
      return store.dispatch(fetchMovies())
        .then(() => {
          expect(store.getActions()).toEqual([
            receiveMovies(movies)
          ]);
          done();
        });
    });

    it('should create fetchIncLike action', (done) => {
      return store.dispatch(fetchIncLike(incLikeMovie.id))
        .then(() => {
          expect(store.getActions()).toEqual([
            incLike(incLikeMovie.id)
          ]);
          done();
        });
    });

    it('should create fetchDecLike action', (done) => {
      return store.dispatch(fetchDecLike(decLikeMovie.id))
        .then(() => {
          expect(store.getActions()).toEqual([
            decLike(decLikeMovie.id)
          ]);
          done();
        });
    });

    it('should create fetchSetRating action', (done) => {
      return store.dispatch(fetchSetRating(movies[0].id, 2))
        .then(() => {
          expect(store.getActions()).toEqual([
            setRating(movies[0].id, 2)
          ]);
          done();
        });
    });

    it('should create fetchDeleteMovie action', (done) => {
      return store.dispatch(fetchDeleteMovie(movies[1].id))
        .then(() => {
          expect(store.getActions()).toEqual([
            deleteMovie(movies[1].id)
          ]);
          done();
        });
    });

    it('should create fetchSaveMovie action', (done) => {
      return store.dispatch(fetchSaveMovie(newMovie))
        .then(() => {
          expect(store.getActions()).toEqual([
            saveMovie(newMovie)
          ]);
          done();
        });
    });
  });

  describe('REDUCER', () => {
    it('should handle initial state', () => {
      expect(movieReducer(undefined, {})).toEqual(initialState);
    });

    it('should handle RECEIVE_ACTORS', () => {
      const resultStore = {
        ...initialState,
        actors,
      };

      expect(
        movieReducer(initialState, receiveActors(actors))
      ).toEqual(resultStore);
    });

    it('should handle RECEIVE_MOVIES', () => {
      const resultStore = {
        ...initialState,
        movies,
      };

      expect(
        movieReducer(initialState, receiveMovies(movies))
      ).toEqual(resultStore);
    });

    it('should handle INCLIKE', () => {
      const resultStore = {
        ...initialState,
        movies: [{
          ...movies[0],
          likes: movies[0].likes + 1,
        }, movies[1]],
        actors,
      };

      expect(
        movieReducer({ ...initialState, movies, actors }, incLike(movies[0].id))
      ).toEqual(resultStore);
    });

    it('should handle DECLIKE', () => {
      const resultStore = {
        ...initialState,
        movies: [{
          ...movies[0],
          likes: movies[0].likes - 1,
        }, movies[1]],
        actors,
      };

      expect(
        movieReducer({ ...initialState, movies, actors }, decLike(movies[0].id))
      ).toEqual(resultStore);
    });

    it('should handle SET_RATING', () => {
      const resultStore = {
        ...initialState,
        movies: [{
          ...movies[0],
          stars: 3,
        }, movies[1]],
        actors,
      };

      expect(
        movieReducer({ ...initialState, movies, actors }, setRating(movies[0].id, 3))
      ).toEqual(resultStore);
    });

    it('should handle DELETE_MOVIE', () => {
      const resultStore = {
        ...initialState,
        movies: [movies[1]],
        actors,
      };

      expect(
        movieReducer({ ...initialState, movies, actors }, deleteMovie(movies[0].id))
      ).toEqual(resultStore);
    });

    it('should handle SAVE_MOVIE', () => {
      const resultStore = {
        ...initialState,
        movies: [newMovie, movies[1]],
        actors,
      };

      expect(
        movieReducer({ ...initialState, movies, actors }, saveMovie(newMovie))
      ).toEqual(resultStore);
    });
  })
});
