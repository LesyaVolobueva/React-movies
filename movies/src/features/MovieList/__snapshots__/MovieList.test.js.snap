// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-- MovieList -- VIEW renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <img
            alt="like"
            className={undefined}
            onClick={[Function]}
            src="https://goo.gl/NSjZGo"
          />
        </div>
        <div
          className={undefined}
        >
          <img
            alt="dislike"
            className={undefined}
            onClick={[Function]}
            src="https://goo.gl/D22jMu"
          />
        </div>
        <span>
          Likes
        </span>
        <div
          className={undefined}
        >
          22
        </div>
      </div>
      <div
        className={undefined}
      >
        <a
          href="movies/1"
          onClick={[Function]}
        >
          <div
            className={undefined}
          >
            The Shawshank Redemptoin
          </div>
        </a>
        <img
          alt="poster"
          className={undefined}
          src="https://images-na.ssl-images-amazon.com/images/M/MV5BODU4MjU4NjIwNl5BMl5BanBnXkFtZTgwMDU2MjEyMDE@._V1_SY500_CR0,0,336,500_AL_.jpg"
        />
      </div>
      <div
        className={undefined}
      >
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
      </div>
    </div>
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          <img
            alt="like"
            className={undefined}
            onClick={[Function]}
            src="https://goo.gl/NSjZGo"
          />
        </div>
        <div
          className={undefined}
        >
          <img
            alt="dislike"
            className={undefined}
            onClick={[Function]}
            src="https://goo.gl/D22jMu"
          />
        </div>
        <span>
          Likes
        </span>
        <div
          className={undefined}
        >
          15
        </div>
      </div>
      <div
        className={undefined}
      >
        <a
          href="movies/4"
          onClick={[Function]}
        >
          <div
            className={undefined}
          >
            The Dark Knight
          </div>
        </a>
        <img
          alt="poster"
          className={undefined}
          src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SY500_CR0,0,337,500_AL_.jpg"
        />
      </div>
      <div
        className={undefined}
      >
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
        <label
          className={undefined}
          onClick={[Function]}
        >
          ★
        </label>
      </div>
    </div>
  </div>
</div>
`;
